spec/services/finance/npq/course_statement_calculator_spec.rb

```diff
@@ -12,6 +12,35 @@
 
   subject { described_class.new(statement:, contract:) }
 
+  describe "#billable_declarations_count_for_declaration_type" do
+    before do
+      declarations = []
+
+      6.times do
+        declarations << create(
+          :npq_participant_declaration,
+          state: "eligible",
+          course_identifier: npq_course.identifier,
+          declaration_type: %w[started retained-1 retained-2 completed].sample,
+        )
+      end
+
+      declarations.each do |dec|
+        Finance::StatementLineItem.create!(
+          statement:,
+          participant_declaration: dec,
+          state: dec.state,
+        )
+      end
+    end
+
+    it "can count different declaration types", :aggregate_failures do
```

Comment:
`aggregate_failures` to save on setup costs therefore time

Reply:
whenever there are multiple `expect` we should enforce adding `:aggregate_failures`. I think  there is a cop for that. I usually do but may skip it sometimes